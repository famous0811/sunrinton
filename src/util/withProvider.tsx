import React from 'react';
import { Provider } from 'mobx-react';

import store from '@src/store';

function withProvider(components: JSX.Element): JSX.Element {
  return <Provider {...store}>{components}</Provider>;
}

export default withProvider;
