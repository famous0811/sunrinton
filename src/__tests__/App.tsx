import React from 'react';
import { render } from '@testing-library/react';

import App from '../App';
import withProvider from '@util/withProvider';

describe('<App/>', () => {
  it('matches snapshot', () => {
    const utils = render(withProvider(<App />));
    expect(utils.container).toMatchSnapshot();
  });
  it('matches value', () => {
    const utils = render(withProvider(<App />));
    utils.getByText('Hello');
    utils.getByText('Hello/World');
  });
});
