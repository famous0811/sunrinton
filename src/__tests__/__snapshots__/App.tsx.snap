// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> matches snapshot 1`] = `
<div>
  <div
    class="sc-bdfBwQ iavMXQ"
  >
    Hello
    <button>
      Hello/World
    </button>
  </div>
</div>
`;
